cumulusci/tasks/metadata/package.py

```diff
@@ -444,3 +539,63 @@ def _run_task(self):
         package_xml = self.package_xml()
         with open(self.options.get("output", output), mode="w", encoding="utf-8") as f:
             f.write(package_xml)
+
+class RemoveSourceComponents(object):
+    def __init__(
+        self,
+        directory,
+        package_xml,
+        api_version
+    ):
+        self.directory = directory
+        self.package_xml = package_xml
+        self.api_version = api_version
+    
+    def __call__(self):
+        xml_map = self.generate_package_xml_map()
+        self.folder_perser = PackageXmlGenerator(
+            directory=self.directory,
+            api_version=self.api_version
+        )
+        self.folder_perser.parse_types()
+        #print(self.folder_perser.types)
+        for parse_type in self.folder_perser.types:
+            parse_type.strip_folder(xml_map[parse_type.metadata_type] if parse_type.metadata_type in xml_map else [])
+        
+
+    def generate_package_xml_map(self) -> map:
+        package_xml_root = elementtree_parse_file(self.package_xml).getroot()
+        xml_map = {}
+        namespace = re.match(r'{.*}', package_xml_root.tag).group(0)
+        for package_type in package_xml_root.findall('{}types'.format(namespace)):
+            type_name = package_type.find('{}name'.format(namespace)).text
+            members = []
+            for package_member in package_type.findall('{}members'.format(namespace)):
+                member_name = package_member.text
+                members.append(member_name)
+            xml_map[type_name] = members
+        return xml_map
+
+class RemoveUnwantedComponents(BaseTask):
+    task_options = {
+        "path": {
+            "description": "The path to a folder of metadata to strip the components",
+            "required": True,
+        },
+        "package_xml": {
+            "description": "The path to package xml file to refer",
+            "required": True,
+        },
+    }
+
+    def _init_options(self, kwargs):
+        super(RemoveUnwantedComponents, self)._init_options(kwargs)
```

Comment:
```suggestion
        super()._init_options(kwargs)
```

We still see this idiom a lot in the code base, but it's a Python 2 idiom that we don't actually need anymore.